pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -291,97 +285,33 @@ func (e *etcd) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, e.client, clientNetworkPolicy, func() error {
-		clientNetworkPolicy.Annotations = map[string]string{
-			v1beta1constants.GardenerDescription: "Allows Ingress to etcd pods from the Shoot's Kubernetes API Server.",
-		}
-		clientNetworkPolicy.Labels = map[string]string{
-			v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
-		}
-		clientNetworkPolicy.Spec.PodSelector = metav1.LabelSelector{
-			MatchLabels: GetLabels(),
-		}
-		clientNetworkPolicy.Spec.Ingress = []networkingv1.NetworkPolicyIngressRule{
-			{
-				From: []networkingv1.NetworkPolicyPeer{
-					{
-						PodSelector: &metav1.LabelSelector{
-							// TODO: Replace below map with a function call to the to-be-introduced kubeapiserver package.
-							MatchLabels: map[string]string{
-								v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
-								v1beta1constants.LabelApp:   v1beta1constants.LabelKubernetes,
-								v1beta1constants.LabelRole:  v1beta1constants.LabelAPIServer,
-							},
-						},
-					},
-					{
-						PodSelector: &metav1.LabelSelector{
-							MatchLabels: monitoring.GetPrometheusLabels(),
-						},
-					},
-				},
-				Ports: []networkingv1.NetworkPolicyPort{
-					{
-						Protocol: &protocolTCP,
-						Port:     &intStrPortEtcdClient,
-					},
-					{
-						Protocol: &protocolTCP,
-						Port:     &intStrPortBackupRestore,
-					},
-				},
-			},
-		}
-		clientNetworkPolicy.Spec.Egress = nil
-		clientNetworkPolicy.Spec.PolicyTypes = []networkingv1.PolicyType{networkingv1.PolicyTypeIngress}
-		return nil
-	}); err != nil {
-		return err
-	}
-
-	// create peer network policy only if the shoot has a HA control plane
-	if e.hasHAControlPlane() {
-		if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, e.client, peerNetworkPolicy, func() error {
-			peerNetworkPolicy.Annotations = map[string]string{
-				v1beta1constants.GardenerDescription: "Allows Ingress to etcd pods from etcd pods for peer communication.",
+	if e.values.Role == v1beta1constants.ETCDRoleMain {
```

Comment:
Why create network policies only for `etcd-main`? Shouldn't `etcd-events` also have the same network policies attached to it?

Reply:
This was easily leading to conflicts because both `etcdMain` and `etcdEvents` component deployers executed this code in the past. However, these network policies are not specified to them (they apply to both because there is no `role` label in the selector). Hence, it doesn't make sense if both component deployers are running this code.

Reply:
Ah ok